### PR TITLE
Bug 1870345: test: extended: only run new machines test against platforms that have machines

### PR DESCRIPTION
**Pull request by sjenning**
@deads2k @sdodson 

https://bugzilla.redhat.com/show_bug.cgi?id=1870345

### CODE REVIEW

**Comment by mfojtik** [comment]
This is blocking metal tests to run, so I will button merge this when the e2e-upgrade yields green.

@sdodson @sjenning any objections?

File: test/extended/machines/cluster.go [hunk 1]
```diff
@@ -12,16 +12,17 @@ import (
...
-	"github.com/openshift/origin/test/extended/util/ibmcloud"
 )
 
 var _ = g.Describe("[sig-cluster-lifecycle][Feature:Machines][Early] Managed cluster should", func() {
 	defer g.GinkgoRecover()
 
 	g.It("have same number of Machines and Nodes", func() {
-		if e2e.TestContext.Provider == ibmcloud.ProviderName {
-			e2eskipper.Skipf("IBM Cloud clusters do not contain machineset resources")
+		if e2e.TestContext.Provider != "aws" &&
```
Comment (cgwalters) [inline test/extended/machines/cluster.go:21]: As I said in a comment in the previous PR, we need to support aws/gcp/etc UPI as well, and I am pretty sure those also don't have machine objects.

So the test should be "is machineAPI enabled"...OK I wrote up https://github.com/openshift/origin/pull/25428

**Pull request merged**